commit c02c4bb2058587d3c012ec08268fd93fdc654ae7
Author: David Brownell <david-b@pacbell.net>
Date:   Fri Jan 20 14:44:12 2006 -0800

    [PATCH] USB: USB authentication states
    
    Another hook needed for wireless USB:  there are states associated with the
    device authentication protocol.  Wireless devices must authenticate using
    the host system's keystore.
    
    Note that wired connections could also use this authentication protocol, if
    for no other reason than to support the most secure "simple" key exchange
    protocols for wireless devices.
    
    Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

diff --git a/include/linux/usb_ch9.h b/include/linux/usb_ch9.h
index ee21e6bf3867..a2aacfc7af2f 100644
--- a/include/linux/usb_ch9.h
+++ b/include/linux/usb_ch9.h
@@ -535,9 +535,11 @@ enum usb_device_state {
 	 */
 	USB_STATE_NOTATTACHED = 0,
 
-	/* the chapter 9 device states */
+	/* chapter 9 and authentication (wireless) device states */
 	USB_STATE_ATTACHED,
-	USB_STATE_POWERED,
+	USB_STATE_POWERED,			/* wired */
+	USB_STATE_UNAUTHENTICATED,		/* auth */
+	USB_STATE_RECONNECTING,			/* auth */
 	USB_STATE_DEFAULT,			/* limited function */
 	USB_STATE_ADDRESS,
 	USB_STATE_CONFIGURED,			/* most functions */